flink-end-to-end-tests/test-scripts/test_sql_client_common.sh

```diff
@@ -17,8 +17,6 @@
 # limitations under the License.
 ################################################################################
 
-set -Eeuo pipefail
```

Comment:
@pnowojski 

> @yanghua please do not trigger rebuild of errors like this:
> 
> ```
> Waiting for Elasticsearch node to work...
> curl: (7) Failed to connect to localhost port 9200: Connection refused
> curl: (7) Failed to connect to localhost port 9200: Connection refused
> 
> [FAIL] 'SQL Client end-to-end test' failed after 0 minutes and 29 seconds! Test exited with exit code 7
> ```
> In the test that you are just adding. This test must be fixed in someway.

I should have found the cause of this problem. We can't enable this setting here, because curl is constantly retrying until it is accessible, and when this setting is added, it will report a connection error.

Another exception is the known FLINK-10801. As a temporary treatment, I will use sleep again to fix it (to quickly verify other parts).

Reply:
For curl it should be able to fix it in the following way:
```
curl -XGET 'http://localhost:9200' || true
```
Like it was fixed in multiple places recently https://github.com/apache/flink/pull/7034 (those were fixes after https://github.com/apache/flink/pull/6924 )